ngx-tools/type-guards/src/is-html-input-element/is-html-input-element.ts

```diff
@@ -0,0 +1,20 @@
+import { isSet } from './../is-set/is-set';
+
+
+/**
+ * Coerce the type to HTMLInputElement
+ *
+ * @param x - The item to test
+ * @return True if the value is a HTMLInputElement
+ *
+ * @example
+ * const myInput = document.querySelector('#myInput');
+ * const myDiv = document.querySelector('#myDiv');
+ *
+ * isHTMLInputElement(myInput); // Returns: true
+ * isHTMLInputElement(myDiv);   // Returns: false
+ */
+// tslint:disable-next-line no-any
+export function isHTMLInputElement(x: any): x is HTMLInputElement {
+  return !!x && isSet(x.files);
```

Comment:
why `.files` here? isn't that  for type file only? (which I think is not true based on the specs, but just wondering....)

Reply:
That's a great question. I was so focused on the task I wasn't really vetting the content of the guards.

I think I used `files` originally because this guard was created to help TypeScript sort out the target element of a drop event within `TsFileUpload`. Originally meant to be private, but pulled out during a reorganization.

We should probably be using `nodeName` instead. If we really need to know that it's a 'file' input, we can chain this check with another that does actually check for files.

Created ticket: https://github.com/getterminus/ngx-tools/issues/325